core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1771,8 +1775,12 @@ public long getNewBlockIdForFile(AlluxioURI path) throws FileDoesNotExistExcepti
          LockedInodePath inodePath = mInodeTree.lockFullInodePath(path, LockPattern.WRITE_INODE);
          FileSystemMasterAuditContext auditContext =
             createAuditContext("getNewBlockIdForFile", path, null, inodePath.getInodeOrNull())) {
+      Mode.Bits permissionNeed = Mode.Bits.WRITE;
+      if (skipFileWritePermissionCheck(inodePath)) {
+        permissionNeed = Mode.Bits.NONE;
```

Comment:
Add a comment to explain why no permission is needed.

Reply:
Added